push/src/main/java/org/aerogear/mobile/push/PushService.java

```diff
@@ -0,0 +1,318 @@
+package org.aerogear.mobile.push;
+
+import static java.net.HttpURLConnection.HTTP_NO_CONTENT;
+import static java.net.HttpURLConnection.HTTP_OK;
+import static org.aerogear.mobile.core.utils.SanityCheck.nonNull;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import com.google.firebase.iid.FirebaseInstanceId;
+import com.google.firebase.messaging.FirebaseMessaging;
+
+import android.content.Context;
+import android.content.pm.ApplicationInfo;
+import android.content.pm.PackageManager;
+import android.os.Bundle;
+import android.os.Handler;
+import android.os.Looper;
+import android.util.Base64;
+
+import org.aerogear.mobile.core.Callback;
+import org.aerogear.mobile.core.MobileCore;
+import org.aerogear.mobile.core.ServiceModule;
+import org.aerogear.mobile.core.configuration.ServiceConfiguration;
+import org.aerogear.mobile.core.exception.HttpException;
+import org.aerogear.mobile.core.http.HttpRequest;
+import org.aerogear.mobile.core.http.HttpResponse;
+
+public class PushService implements ServiceModule {
+
+    public static final String DEFAULT_MESSAGE_HANDLER_KEY = "DEFAULT_MESSAGE_HANDLER_KEY";
+
+    private static final String registryDeviceEndpoint = "rest/registry/device";
+    private static final String JSON_OBJECT = "android";
+    private static final String JSON_VARIANT_ID = "variantId";
+    private static final String JSON_VARIANT_SECRET = "variantSecret";
+    private static final String JSON_SENDER_ID = "senderId";
+
+    private static final List<MessageHandler> MAIN_THREAD_HANDLERS = new ArrayList<>();
+    private static final List<MessageHandler> BACKGROUND_THREAD_HANDLERS = new ArrayList<>();
+
+    private final String deviceType = "ANDROID";
+    private final String operatingSystem = "android";
+    private final String osVersion = android.os.Build.VERSION.RELEASE;
+
+    private MobileCore core;
+    private String url;
+    private UnifiedPushCredentials unifiedPushCredentials;
+
+    private static MessageHandler defaultHandler;
+
+    @Override
+    public String type() {
+        return "push";
+    }
+
+    @Override
+    public void configure(MobileCore core, ServiceConfiguration serviceConfiguration) {
+        this.core = core;
+        this.url = serviceConfiguration.getUrl();
+        getDefaultHandler(core.getContext());
+
+        try {
+            JSONObject android =
+                            new JSONObject(serviceConfiguration.getProperties().get(JSON_OBJECT));
```

Comment:
instead of `JSON_OBJECT` would `ANDROID_CONFIG_KEY` be a better name?

Reply:
I have updated it to:

```
private static final String JSON_ANDROID_CONFIG_KEY = "android";
private static final String JSON_VARIANT_ID_KEY = "variantId";
private static final String JSON_VARIANT_SECRET_KEY = "variantSecret";
private static final String JSON_SENDER_ID_KEY = "senderId";
```